yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -130,6 +131,13 @@ type ParserIssueDetector struct {
 	// Indexes stored per table
 	// Key is table name, value is slice of indexes on that table.
 	tableIndexes map[string][]*queryparser.Index
+
+	// Primary key columns by table (qualified table name)
+	tablePrimaryKeys map[string][]string
+	// Unique constraint columns by table (list of column sets)
+	tableUniqueConstraints map[string][][]string
+	// NOT NULL columns by table
+	tableNotNullColumns map[string]map[string]bool
```

Comment:
As discussed, let's create a TableMetadata struct that stores all the relevant information for issue detection rather than creating a map for every use-case. 

rough notes from discussion: 
```
map[table_name]Table

struct Table:
	hasPk : true/false
	constraints: []
	indexes: []
	columns: 
		column1:
			datatype: varchar
			IsNotNull: true
	constraints: []

getPkRecommendation:
	for table in tables:
		if not table.hasPk:
			candidatePKs := []
			for index in table.indexes:
				// check if unique index
				// check if all are  not null. 

*/
// type TableMetadata struct {
// 	HasPk bool
// 	PkColumnNames []string
// 	Columns []ColumnMetadata
// 	Constraints []ConstraintMetadata
// 	NotNullColumns []string
// }
```